backend/src/folder/folder.service.ts

```diff
@@ -0,0 +1,80 @@
+import { Injectable } from '@nestjs/common';
+import { CreateFolderInput } from './dto/create-folder.input';
+import { UpdateFolderInput } from './dto/update-folder.input';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { Folder } from './entities/folder.entity';
+
+@Injectable()
+export class FolderService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async getFolders(userId: string): Promise<Folder[]> {
+    return this.prisma.folder.findMany({
+      where: { userId },
+    });
+  }
+
+  async getFolder(id: string, userId: string): Promise<Folder> {
+    const folder = await this.prisma.folder.findUnique({
+      where: { id, userId },
+      include: { journals: true } as any,
```

Comment:
Since your Folder Entity type has journals as an array, a better way could be to add a resolver for that field

Right now, your `getFolders` mutation will not return journals for the folder even if you request it

And this `getFolder` function will always get journals from the database, even if you don't ask for it

In this case, graphql has a useful concept of field resolvers 

You can take a look at the example in here: https://docs.nestjs.com/graphql/resolvers#code-first-resolver

Basically what you will do is the following:
`folderService.getFolders` will not include journals and it will not return them

In folders resolver you can create a separate field resolver for journals
It can look something like this:
```
@ResolveField()
async journals(@Parent() folder: Folder) {
  const { id } = folder;
  return this.journalsService.getJournalsByFolderId(id);
}

What this will do is whenever you will ask for journals field in your queries for folder (both getJournal and getJournals) — it will run this resolve field function and get all the journals for your folders.
If you won't ask for journals in your query — it will not make this database request and only return the folder without journals
```

Reply:
You don't really have to do this, but it can be a nice way of doing that — this way you can get all journals not only in `getFolder` but also in `getFolders` queries

But you don't have to change it this way and you can keep it the way you already have if you want that

What do you think?

Reply:
Done!